### ISSUE TITLE
RedundantModifier should include RECORD_DEF in acceptable tokens

### ISSUE THREAD

**Issue by pugnascotia** [checkstyle/checkstyle#11259]
Issue property reported by @Vyom-Yadav

I have read check documentation: http://checkstyle.sourceforge.net/config_modifier.html#RedundantModifier
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RedundantModifier"/>
    </module>
</module>

/var/tmp $ cat Test.java
final record foo() {

    static final record hoo() {
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml YOUR_FILE.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml YOUR_FILE.java
Starting audit...
[ERROR] /var/tmp/Test.java:1:1: Redundant 'final' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:3:5: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:3:12: Redundant 'final' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 3 errors.
```

Related info https://docs.oracle.com/en/java/javase/17/language/records.html

**Comment by strkkk** [comment]
related info https://docs.oracle.com/en/java/javase/17/language/records.html

**Comment by Vyom-Yadav** [comment]
I am on it.

**Comment by nmancus1** [comment]
It would be a good idea to wait for resolution of https://github.com/checkstyle/checkstyle/issues/10798 before working on this issue.

**Comment by Vyom-Yadav** [comment]
@nmancus1 Though both the issues are related to the same check, I feel that we can go ahead with this issue as this one isn't dependent on the java version (afaik) unlike the other issue.

**Comment by strkkk** [comment]
> It would be a good idea to wait for resolution of #10798 before working on this issue.

I do not think it is necessary, this issue can be fixed separately.

**Comment by Vyom-Yadav** [comment]
Records are also implicitly final, will be fixed in the same PR.

**Comment by Vyom-Yadav** [comment]
I have read check documentation: http://checkstyle.sourceforge.net/config_modifier.html#RedundantModifier
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RedundantModifier"/>
    </module>
</module>

/var/tmp $ cat Test.java
final record foo() {

    static final record hoo() {
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml YOUR_FILE.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml YOUR_FILE.java
Starting audit...
[ERROR] /var/tmp/Test.java:1:1: Redundant 'final' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:3:5: Redundant 'static' modifier. [RedundantModifier]
[ERROR] /var/tmp/Test.java:3:12: Redundant 'final' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 3 errors.
```

Related info https://docs.oracle.com/en/java/javase/17/language/records.html
